checker/src/main/java/org/checkerframework/checker/mustcall/CreatesMustCallForToJavaExpression.java

```diff
@@ -0,0 +1,201 @@
+   * @param tree a method declaration
+   * @param atypeFactory the type factory to report errors and parse the expression string
+   * @param supplier a type factory that can supply the executable elements for CreatesMustCallFor
+   *     and CreatesMustCallFor.List's value elements. Usually, you should just pass atypeFactory
+   *     again. The arguments are different so that the given type factory's adherence to both
+   *     protocols are checked by the type system.
+   * @return the arguments of the method's @CreatesMustCallFor annotation, or an empty list
+   */
+  public static List<JavaExpression> getCreatesMustCallForExpressionsAtMethodDeclaration(
+      MethodTree tree,
+      GenericAnnotatedTypeFactory<?, ?, ?, ?> atypeFactory,
+      CreatesMustCallForElementSupplier supplier) {
+    List<JavaExpression> results = new ArrayList<>(1);
+    ExecutableElement method = TreeUtils.elementFromDeclaration(tree);
+    List<AnnotationMirror> createsMustCallForAnnos =
+        getCreatesMustCallForAnnos(method, atypeFactory, supplier);
+    for (AnnotationMirror createsMustCallFor : createsMustCallForAnnos) {
+      JavaExpression expr =
+          getCreatesMustCallForExpression(
+              createsMustCallFor,
+              tree,
+              method.getSimpleName(),
+              atypeFactory,
+              supplier,
+              (s) -> StringToJavaExpression.atMethodBody(s, tree, atypeFactory.getChecker()));
+      if (expr != null && !results.contains(expr)) {
+        results.add(expr);
+      }
+    }
+    return results;
+  }
+
+  /**
+   * Returns the {@code CreatesMustCallFor} annotations on a method
+   *
+   * @param method the method
+   * @param atypeFactory the type factory to use for looking up annotations
+   * @param supplier supplier to use to get elements
+   * @return the {@code CreatesMustCallFor} annotations
+   */
+  private static List<AnnotationMirror> getCreatesMustCallForAnnos(
+      ExecutableElement method,
+      GenericAnnotatedTypeFactory<?, ?, ?, ?> atypeFactory,
+      CreatesMustCallForElementSupplier supplier) {
+    AnnotationMirror createsMustCallForList =
+        atypeFactory.getDeclAnnotation(method, CreatesMustCallFor.List.class);
+    List<AnnotationMirror> result = new ArrayList<>();
+    if (createsMustCallForList != null) {
+      // Handle a set of CreatesMustCallFor annotations.
+      result.addAll(
+          AnnotationUtils.getElementValueArray(
+              createsMustCallForList,
+              supplier.getCreatesMustCallForListValueElement(),
+              AnnotationMirror.class));
+    }
+    AnnotationMirror createsMustCallFor =
+        atypeFactory.getDeclAnnotation(method, CreatesMustCallFor.class);
+    if (createsMustCallFor != null) {
+      result.add(createsMustCallFor);
+    }
+    return result;
+  }
+
+  /**
+   * Parses a single CreatesMustCallFor annotation. Clients should use {@link
+   * #getCreatesMustCallForExpressionsAtInvocation(MethodInvocationNode,
```

Comment:
I don't think this suggestion is relevant or even necessarily true anymore:
* it's not relevant, since this method is now private rather than package-private, so no clients have access to it (this is a good change you orchestrated by moving these methods out of that interface!)
* it's not longer really true, since the interface of this method vs `getCreatesMustCallForExpressionsAtInvocation` is now different, and that method is no longer the only alternative (`getCreatesMustCallForExpressionsAtDeclaration` now exists, too)

I think we can just remove this suggestion, since it's clear that clients of this class should use the `public` methods.

Reply:
3fc29f3b6a626edf9b37c734b773d062dcdc52d0